File: src/main/java/br/com/leonardoferreira/jirareport/client/HolidayClient.java
package br.com.leonardoferreira.jirareport.client;

import java.util.List;

import br.com.leonardoferreira.jirareport.domain.vo.HolidayVO;
import org.springframework.cache.annotation.Cacheable;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

/**
 * Created by jfalbo
 */
@FeignClient(name = "holiday-client", url = "${holiday.url}")
public interface HolidayClient {

    @GetMapping
    @Cacheable("findAllHolidaysInCity")
    List<HolidayVO> findAllHolidaysInCity(@RequestParam("ano") Integer year,
                                          @RequestParam("estado") String state,
                                          @RequestParam("cidade") String city,
                                          @RequestParam("token") String token);
}

File: client-library/src/main/kotlin/br/com/jiratorio/client/HolidayClient.kt
package br.com.jiratorio.client

import br.com.jiratorio.config.HolidayClientConfig
import br.com.jiratorio.domain.response.holiday.HolidayApiResponse
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam

@FeignClient(
    name = "holiday-client",
    url = "\${holiday.url}",
    configuration = [
        HolidayClientConfig::class
    ]
)
interface HolidayClient {

    @GetMapping
    fun findAllHolidaysInCity(
        @RequestParam("ano") year: Int,
        @RequestParam("estado") state: String,
        @RequestParam("cidade") city: String,
        @RequestParam("token") token: String
    ): List<HolidayApiResponse>

}
